contracts/PeeranhaNFT.sol

```diff
@@ -26,9 +27,26 @@ contract PeeranhaNFT is IPeeranhaNFT, ChildMintableERC721Upgradeable {
   }
 
   function _beforeTokenTransfer(address from, address to, uint256 amount) internal virtual override (ChildMintableERC721Upgradeable) {
+    AchievementCommonLib.AchievementsType nftType = getNftType(amount);
+    if (nftType == AchievementCommonLib.AchievementsType.SoulRating) {
+      require(from == address(0) || to == address(0), "You_can_not_transfer_soul_bound");
+    }
+
     super._beforeTokenTransfer(from, to, amount);
   }
 
+  function getNftType(
```

Comment:
`getNftType` -> `getAchievementsType`
`NftId` -> `achievementsType`
to be consistent with names everywhere else.